Index: src/serviceWorker.js
===================================================================
--- src/serviceWorker.js	(revision 5f2f23fd7eeb7051b58b39bbed29e8bc8c2a48ec)
+++ services/app/src/serviceWorker.js	(date 1559114359000)
@@ -21,7 +21,8 @@
 );
 
 export function register(config) {
-  if ('serviceWorker' in navigator) {
+  // if ('serviceWorker' in navigator) {
+    if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
     // The URL constructor is available in all browsers that support SW.
     const publicUrl = new URL(process.env.PUBLIC_URL, window.location.href);
     if (publicUrl.origin !== window.location.origin) {
